src/ITensorChainRules/ITensorChainRules.jl

```diff
@@ -326,6 +319,52 @@ function ChainRulesCore.rrule(::typeof(apply), x1::Vector{ITensor}, x2::MPS; kwa
   return y, apply_pullback
 end
 
+function ChainRulesCore.rrule(::typeof(apply), x1::Vector{ITensor}, x2::MPO; kwargs...)
+  apply_dag = get(kwargs, :apply_dag, true)
```

Comment:
Shouldn't this default to `false` since the original `apply` function defaults to `false`?

You may be able to not set a default at all here, could you try just using `kwargs.apply_dag`? Basically my concern is that this keyword argument should match whatever was set by the user for the forward pass of `apply`, so we want to make sure it matches that and not accidentally set it to something different.